### PR TITLE
[DST-367]: rac `SelectList` component

### PR DESCRIPTION
**Pull request by aromko**
In order to solve the multiselect issue from TCO it could be a possible solution to use the [gridlist](https://react-spectrum.adobe.com/react-aria/GridList.html) from react-aria. It's similiar to a listbox but offers some other stuf, like action button which is maybe useful in future.

### CODE REVIEW

**Comment by sarahgm** [comment]
is this still a draft?

**Comment by aromko** [comment]
> @aromko is this still a draft?

@sarahgm Yes. It's just the first try from our mob. We can decide in our next cycle when we continue with that or feel free to go on 😄

File: docs/content/components/collection/gridlist/selectlist-multiple-selection.demo.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      >
+        {(item: { id: any; name: string }) => (
+          <SelectList.Item id={item.id}>
+            {item.name}
+            <Button
+              aria-label="Info"
+              onPress={() => alert(`Info for ${item.name}...`)}
+              className="ml-auto"
+            >
+              ⓘ
```
Comment (sarahgm) [inline docs/content/components/collection/gridlist/selectlist-multiple-selection.demo.tsx:31]: don't we have an own icon for it?

File: packages/components/src/SelectList/SelectListItem.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    <SelectListItem
+      textValue={textValue}
+      {...props}
+      className={cn('flex items-center', classNames?.option)}
+      ref={ref as Ref<HTMLDivElement>}
+    >
+      {({ selectionMode, selectionBehavior, allowsDragging }) => (
+        <>
+          {/* Add elements for drag and drop and selection. */}
+          {allowsDragging && <Button slot="drag">≡</Button>}
```
Comment (sarahgm) [inline packages/components/src/SelectList/SelectListItem.tsx:29]: do we need this?

**Review by sarahgm — CHANGES_REQUESTED** [review]

File: packages/components/src/SelectList/SelectListItem.tsx (hunk 2 above)
Comment (sarahgm) [reply packages/components/src/SelectList/SelectListItem.tsx:29]: or other: do we want to support this?

File: docs/content/components/collection/gridlist/selectlist-multiple-selection.demo.tsx (hunk 1 above)
Comment (OsamaAbdellateef) [reply docs/content/components/collection/gridlist/selectlist-multiple-selection.demo.tsx]: @sarahgm   i added a new one, check it plz

**Comment by aromko** [comment]
@OsamaAbdellateef Can't approve because I'm the creator of the PR. But it's fine for me now.

**Comment by OsamaAbdellateef** [comment]
> @OsamaAbdellateef Can't approve because I'm the creator of the PR. But it's fine for me now.
Then @sebald  or @sarahgm  can merge :smile:

**Review by sarahgm — APPROVED** [review]

**Pull request merged**